config/src/network_id.rs

```diff
@@ -89,6 +89,39 @@ pub enum NetworkId {
     Private(String),
 }
 
+impl Ord for NetworkId {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.partial_cmp(other).unwrap()
+    }
+}
+
+impl PartialOrd for NetworkId {
+    /// Generalized ordering for determing which network is the most important
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        // To simplify logic below, if it's the same it's equal
+        Some(if self == other {
+            Ordering::Equal
+        } else {
+            // Everywhere below assumes that equal has already been covered
+            match self {
+                NetworkId::Validator => Ordering::Less,
+                NetworkId::Public => Ordering::Greater,
+                NetworkId::Private(_) => match other {
+                    NetworkId::Validator => Ordering::Greater,
+                    NetworkId::Public => Ordering::Less,
+                    NetworkId::Private(_) => {
+                        if self.is_vfn_network() {
```

Comment:
Honestly, I think the original usage of this `NetworkId::Private` was too forward thinking and too short sighted at the same time.  It basically causes me to do this `is_vfn_network` everywhere, but it's being serialized like this, so it's hard to change now.

I'm pretty sure it'll be a tiebreaker, so if I have to networks "a" and "b", then it'll only do one comparison, and whichever came first in the iterator will be marked as `less`.

Let's also remember that there shouldn't be a `vfn1` and a `vfn2` because that'll be confusing.  It would be for a completely different private network that we haven't even imagined yet, which is what is causing a ton of confusion.  You could use your own private network, and not use the public network if you say wanted to connect your full nodes together and ignore everyone else.